src/charm.py

```diff
@@ -19,6 +29,230 @@ class KratosCharm(CharmBase):
+            self.unit.status = BlockedStatus("Waiting for database creation")
+            return
+
+        self._update_kratos_config_file()
+
+        try:
+            self._update_layer()
+            self.unit.status = ActiveStatus()
+        except ErrorWithStatus as e:
+            self.model.unit.status = e.status
+            if isinstance(e.status, BlockedStatus):
+                logger.error(str(e.msg))
+            else:
+                logger.info(str(e.msg))
+
+        self._run_sql_migration()
+
+    def _run_sql_migration(self) -> None:
+        """Runs a command to create SQL schemas and apply migration plans."""
+        process = self._container.exec(
+            ["kratos", "migrate", "sql", "-e", "--config", self._config_file_path, "--yes"],
+            timeout=20.0,
+        )
+        try:
+            stdout, _ = process.wait_output()
+            logger.info(f"Executing automigration: {stdout}")
+        except ExecError as err:
+            logger.error(f"Exited with code {err.exit_code}. Stderr: {err.stderr}")
+            self.unit.status = BlockedStatus("Database migration job failed")
+
+    def _on_install(self, event) -> None:
+        """Event Handler for install event.
+
+        - push configs
+        - apply service manifests
+        - update pod template to expose 2 ports
+        """
+        self.unit.status = MaintenanceStatus("Configuring/deploying resources")
+
+        if not self.unit.is_leader():
+            event.defer()
+            logger.info("Waiting for leadership to apply k8s resources")
+            self.unit.status = WaitingStatus("Waiting for leadership to apply k8s resources")
+            return
+
+        try:
+            self.resource_handler.apply()
+
+        except (ApiError, ErrorWithStatus) as e:
+            if isinstance(e, ApiError):
+                logger.error(
+                    f"Applying resources failed with ApiError status code {e.status.code}, error response: {e.response}"
+                )
+                self.unit.status = BlockedStatus(f"ApiError: {e.status.code}")
+            else:
+                logger.info(e.msg)
+                self.unit.status = e.status
+        else:
+            self.unit.status = ActiveStatus()
+
+    def _on_pebble_ready(self, event) -> None:
+        """Event Handler for pebble ready event.
+
+        Do the following if kratos container can be connected:
+        - push configs
+        - update pebble layer.
+        """
+        self.unit.status = MaintenanceStatus("Configuring/deploying resources")
+
+        self._update_container(event)
+
+    def _on_database_created(self, event: DatabaseCreatedEvent) -> None:
+        # Handle the created database
+        # Create configuration file for app
+        self.unit.status = MaintenanceStatus("Retrieving database details")
+
+        self._update_container(event)
+
+        self._run_sql_migration()
```

Comment:
I believe you don't need to run the migration here since it's already called in `_update_container`
```suggestion
```